api/src/pcapi/core/educational/api.py

```diff
@@ -703,3 +707,62 @@ def _get_expired_collective_offer_ids(interval: list[datetime.datetime], page: i
     collective_offers = educational_repository.get_expired_collective_offers(interval)
     collective_offers = collective_offers.offset(page * limit).limit(limit)
     return [offer_id for offer_id, in collective_offers.with_entities(educational_models.CollectiveOffer.id)]
+
+
+def list_collective_offers_for_pro_user(
+    user_id: int,
+    user_is_admin: bool,
+    category_id: Optional[str],
+    offerer_id: Optional[int],
+    venue_id: Optional[int] = None,
+    name_keywords_or_isbn: Optional[str] = None,
+    status: Optional[str] = None,
+    creation_mode: Optional[str] = None,
+    period_beginning_date: Optional[str] = None,
+    period_ending_date: Optional[str] = None,
+) -> list[Union[CollectiveOffer, CollectiveOfferTemplate]]:
+    offers = get_collective_offers_for_filters(
+        user_id=user_id,
+        user_is_admin=user_is_admin,
+        offers_limit=OFFERS_RECAP_LIMIT,
+        offerer_id=offerer_id,
+        status=status,
+        venue_id=venue_id,
+        category_id=category_id,
+        name_keywords_or_isbn=name_keywords_or_isbn,
+        period_beginning_date=period_beginning_date,
+        period_ending_date=period_ending_date,
+    )
+    templates = get_collective_offers_template_for_filters(
+        user_id=user_id,
+        user_is_admin=user_is_admin,
+        offers_limit=OFFERS_RECAP_LIMIT,
+        offerer_id=offerer_id,
+        status=status,
+        venue_id=venue_id,
+        category_id=category_id,
+        name_keywords_or_isbn=name_keywords_or_isbn,
+        period_beginning_date=period_beginning_date,
+        period_ending_date=period_ending_date,
+    )
+    offer_id = 0
+    template_id = 0
+    merged_offers = []
+    epoch = datetime.datetime(year=1970, month=1, day=1)
+
+    # merge two ordered lists to one shorter than OFFERS_RECAP_LIMIT items
+    for _ in range(min(OFFERS_RECAP_LIMIT, (len(offers) + len(templates)))):
+        offer_date = offers[offer_id].dateCreated if offer_id < len(offers) else epoch
+        template_date = templates[template_id].dateCreated if template_id < len(templates) else epoch
+
+        if offer_id < len(offers) and offer_date > template_date:
+            merged_offers.append(offers[offer_id])
+            offer_id += 1
+        elif template_id < len(templates):
+            merged_offers.append(templates[template_id])
+            template_id += 1
+        else:
+            # this should never hapen. Only there as defensive mesure.
+            break
```

Comment:
Proposition, c'est un peu plus long mais ça a l'avantage d'expliciter les branchements sans passer par une MIN_DATE :)
 
```suggestion
    for _ in range(min(OFFERS_RECAP_LIMIT, (len(offers) + len(templates)))):
        if offer_id < len(offers) and template_id < len(templates):
            # this should never hapen. Only there as defensive mesure.
            break

        if template_id < len(templates):
            merged_offers.append(offers[offer_id])
            continue

        if offer_id < len(offers):
            merged_offers.append(templates[template_id])
            continue

        offer_date = offers[offer_id].dateCreated
        template_date = templates[template_id].dateCreated

        if offer_date > template_date:
            merged_offers.append(offers[offer_id])
            offer_id += 1
        else:
            merged_offers.append(templates[template_id])
            template_id += 1```

Reply:
Il faut u'on en parle. Pour moi ton code ne marche pas

Reply:
j'ai modifié l'algo en m'inspirant de ce que tu proposait, dis moi si ça te va